gatherers/manifest.js

```diff
@@ -54,34 +38,87 @@ class Manifest extends Gather {
     };
   }
 
+  /**
+   * @private
+   */
+  static _createDeferred() {
+    let resolve;
+    let reject;
+    const promise = new Promise((internalResolve, internalReject) => {
+      resolve = internalResolve;
+      reject = internalReject;
+    });
+
+    return {
+      promise,
+      resolve,
+      reject
+    };
+  }
+
+  /**
+   * Creates a listener for manifests in network requests, a load timeout, and
+   * a promise they resolve when complete.
+   * @private
+   */
+  static _resolveOnManifest() {
+    const manifestDeferred = Manifest._createDeferred();
+
+    setTimeout(_ => manifestDeferred.resolve(null), MANIFEST_LOAD_TIMEOUT);
+
+    const manifestListener = function(request) {
+      const resourceType = request.resourceType();
+      if (resourceType && resourceType.name() === 'manifest') {
+        manifestDeferred.resolve(request);
+      }
+    };
+
+    return {
+      listener: manifestListener,
+      promise: manifestDeferred.promise
+    };
+  }
+
   static gather(options) {
     const driver = options.driver;
-    /**
-     * This re-fetches the manifest separately, which could
-     * potentially lead to a different asset. Using the original manifest
-     * resource is tracked in issue #83
-     */
-    return driver.querySelector('head link[rel="manifest"]')
-      .then(node => {
-        if (!node) {
-          return this._errorManifest('No <link rel="manifest"> found in DOM.');
+
+    const manifestLoad = Manifest._resolveOnManifest();
+
+    return driver.beginNetworkCollect(manifestLoad.listener)
+      // Trigger a fetch of the manifest by Chrome.
+      .then(_ => driver.sendCommand('Page.requestAppBanner'))
+
+      // Wait on grabbing the finished manifest network request (or timing out).
+      .then(_ => manifestLoad.promise)
+
+      .then(manifestRequest => {
+        if (!manifestRequest) {
+          return Manifest._errorManifest('Timed out waiting for manifest to load.');
```

Comment:
would be nice to name this something like.. Manifest.FAULT_STATE_REACHED()

also the ._ is kinda silly since we are calling it essentially externally :u5272: